_posts/2021-12-06-Rails-7-0-rc-1-released.markdown

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: 'Rails 7.0 RC1: New JavaScript Answers, At-Work Encryption, Query Origin Logging, Zeitwerk Exclusively'
+categories: releases
+author: dhh
+published: true
+date: 2021-12-06 16:50:00 -05:00
+---
+We're almost ready to declare Rails 7 done! The feedback since [the first alpha release](https://weblog.rubyonrails.org/2021/9/15/Rails-7-0-alpha-1-released/) has been wonderful, we've eliminated a slew of issues, and we've seen Basecamp, HEY, GitHub, and Shopify all run in production on this alpha series. So we now feel so confident that this is nearly ready that we're skipping straight from alpha to release candidate. This is RC1.
+
+Please help us do the final testing of all this new stuff so we can ensure a solid final release of Rails 7 this year!
+
+## All New Answers On The Front-End
+
+After almost five years with [Webpacker](https://github.com/rails/webpacker) as our default answer to writing modern JavaScript in Rails, [it's time to move on](https://www.youtube.com/watch?v=PtxZvFnL2i0). Advancements in browser support for ES6/ESM, widespread adoption of HTTP/2, and [the exciting new standard for import maps](https://github.com/rails/importmap-rails) has [paved the way](https://world.hey.com/dhh/modern-web-apps-without-javascript-bundling-or-transpiling-a20f2755) for [a no-Node approach to JavaScript in Rails 7](https://world.hey.com/dhh/rails-7-will-have-three-great-answers-to-javascript-in-2021-8d68191b) – [without giving up on npm packages](https://github.com/rails/importmap-rails#using-node-modules-via-javascript-cdns).
```

Comment:
@dhh @rafaelfranca The last link here isn't working as intended - I've opened https://github.com/rails/weblog/pull/272 to fix the hash segment.